erpnext/assets/doctype/asset/asset.py

```diff
@@ -453,7 +453,7 @@ def make_gl_entries(self):
 
 		if ((self.purchase_receipt \
 			or (self.purchase_invoice and frappe.db.get_value('Purchase Invoice', self.purchase_invoice, 'update_stock')))
-			and self.purchase_receipt_amount and self.available_for_use_date <= nowdate()):
+			and self.purchase_receipt_amount and getdate(self.available_for_use_date) <= nowdate()):
```

Comment:
also check if `available_for_use_date` is set. `getdate()` returns current date if `None` is passed. Please fix and reopen. Closing this for now